test/workflows/SwapWorkflow.t.sol

```diff
@@ -0,0 +1,179 @@
+import {SwapWorkflow} from "../../src/access/workflows/SwapWorkflow.sol";
+import {WethWorkflow} from "../../src/access/workflows/WethWorkflow.sol";
+import {IAccessPoint} from "../../src/interfaces/IAccessPoint.sol";
+import {IAccessRegistry} from "../../src/interfaces/IAccessRegistry.sol";
+import {IKintoEntryPoint} from "../../src/interfaces/IKintoEntryPoint.sol";
+import {SignaturePaymaster} from "../../src/paymasters/SignaturePaymaster.sol";
+
+import {AccessRegistryHarness} from "../harness/AccessRegistryHarness.sol";
+
+import {UserOp} from "../helpers/UserOp.sol";
+import {UUPSProxy} from "../helpers/UUPSProxy.sol";
+import {SharedSetup} from "../SharedSetup.t.sol";
+
+contract SwapWorkflowTest is UserOp, SharedSetup {
+    using MessageHashUtils for bytes32;
+    using stdJson for string;
+
+    IKintoEntryPoint entryPoint;
+    AccessRegistry internal accessRegistry;
+    IAccessPoint internal accessPoint;
+    SwapWorkflow internal swapWorkflow;
+    WethWorkflow internal wethWorkflow;
+
+    address internal constant EXCHANGE_PROXY = 0xDef1C0ded9bec7F1a1670819833240f027b25EfF;
+    address internal constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address internal constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address internal constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+    function setUp() public override {
+        super.setUp();
+
+        if (!fork) return;
+
+        string memory rpc = vm.envString("ETHEREUM_RPC_URL");
+        require(bytes(rpc).length > 0, "ETHEREUM_RPC_URL is not set");
+
+        vm.chainId(1);
+        mainnetFork = vm.createFork(rpc);
+        vm.selectFork(mainnetFork);
+        assertEq(vm.activeFork(), mainnetFork);
+
+        vm.deal(_owner, 100 ether);
+
+        deploy();
+
+        vm.label(EXCHANGE_PROXY, "EXCHANGE_PROXY");
+    }
+
+    function deploy() internal {
+        entryPoint = IKintoEntryPoint(address(new EntryPoint{salt: 0}()));
+
+        // use random address for access point implementation to avoid circular dependency
+        UpgradeableBeacon beacon = new UpgradeableBeacon(address(this), address(this));
+        IAccessRegistry accessRegistryImpl = new AccessRegistryHarness(beacon);
+        UUPSProxy accessRegistryProxy = new UUPSProxy{salt: 0}(address(accessRegistryImpl), "");
+
+        accessRegistry = AccessRegistry(address(accessRegistryProxy));
+        beacon.transferOwnership(address(accessRegistry));
+        IAccessPoint accessPointImpl = new AccessPoint(entryPoint, accessRegistry);
+
+        accessRegistry.initialize();
+        accessRegistry.upgradeAll(accessPointImpl);
+        accessPoint = accessRegistry.deployFor(address(_user));
+        vm.label(address(accessPoint), "accessPoint");
+
+        swapWorkflow = new SwapWorkflow(EXCHANGE_PROXY);
+        wethWorkflow = new WethWorkflow(address(WETH));
+
+        entryPoint.setWalletFactory(address(accessRegistry));
```

Comment:
Should we do a real test without setting the access registry as the wallet factory?

That's not the real wallet factory. In the Kinto L2, we shouldnt need the access registry or access point right?

The wallet should be the access point on Kinto L2
The wallet factory should be access Registry on Kinto L2.

Reply:
The only reason it sets wallet factory is use test helper which are written for Kinto. This is not needed on other chains. Will fix this during tests refactor.